src/components/BlogPage/BlogPage.js

```diff
@@ -15,29 +19,47 @@ class BlogPage extends React.Component {
     super(props);
 
     this.state = {
+      isLoading: false,
+      category: '',
       currentRangeStart: 0,
       currentRangeEnd: 3,
       isButtonDisabled: false
     };
 
     this.getOlderPosts = this.getOlderPosts.bind(this);
+    this.setCategory = this.setCategory.bind(this);
   }
 
   componentDidMount() {
     this.getBlogItems();
   }
 
-  async getBlogItems() {
+  componentDidUpdate(prevProps, { category: prevCategory }) {
+    const { category } = this.state;
+
+    category !== prevCategory && this.getBlogItems(category);
+  }
+
+  async getBlogItems(category) {
+    let query = '';
+    this.setState({ isLoading: true });
+    if (category) {
+      query = `?category=${category}`;
+    }
+
     const userAPI = new HttpService();
-    const blogsURL = `${process.env.SERVER_URL}/blogs`;
+    const blogsURL = `${process.env.SERVER_URL}/blogs${query}`;
     const { getBlogItemsSuccess, getBlogItemsError } = this.props;
 
     try {
       const response = await userAPI.get(blogsURL);
 
+      setTimeout(() => this.setState({ isLoading: false }), 500);
```

Comment:
Why are you using setTimeout for state update?

Reply:
Because I want to take some time showing preloader

Reply:
Because I want to take some time showing preloader

Reply:
Well, can you move to function then, and give it a meaningful name?